www/assets/js/components/SearchPage.test.tsx

```diff
@@ -14,18 +14,19 @@ import SearchPage from "./SearchPage"
 import { LIST_UI_PAGE_SIZE, COMPACT_UI_PAGE_SIZE } from "../lib/constants"
```

Comment:
The changes to this file are all because SearchPage used to use window.location directly, whereas now it uses a `history` object from React Router's history package. (The history package was already required; no new dependencies have been added).

Anyway, I had to update all the tests to use history instead of `window.location`.

And while I was doing that, I:
- added [jest-watch-typeahead](https://github.com/jest-community/jest-watch-typeahead) for easier filtering of jest tests
- made various ts improvements, e.g., using `spySearch` instead of `search`.